goutubedl_test.go

```diff
@@ -285,6 +285,41 @@ func TestSubtitles(t *testing.T) {
 	}
 }
 
+func TestDownloadSections(t *testing.T) {
+	defer leakChecks(t)()
+
+	cmd := exec.Command("ffmpeg", "-version")
+	_, err := cmd.Output()
+	if err != nil {
+		t.Errorf("failed to check ffmpeg installed: %s", err)
+	}
+
+	ydlResult, ydlResultErr := goutubedl.New(
+		context.Background(),
+		"https://www.youtube.com/watch?v=OyuL5biOQ94",
+		goutubedl.Options{
+			DownloadSections: "*0:0-0:5",
+		})
+
+	
+	if ydlResult.Options.DownloadSections != "*0:0-0:5" {
+		t.Errorf("failed to setup --download-sections")
+	}
+	if ydlResultErr != nil {
+		t.Errorf("failed to download: %s", ydlResultErr)
+	}
+	dr, err := ydlResult.Download(context.Background(), "best")
+	if err != nil {
+		t.Fatal(err)
+	}
+	downloadBuf := &bytes.Buffer{}
+	_, err = io.Copy(downloadBuf, dr)
+	if err != nil {
+		t.Fatal(err)
+	}
+	dr.Close()
```

Comment:
Thinking it would be great to somehow check that `DownloadSections` works, maybe can ffprobe the result?
```
$ yt-dlp --download-sections '*0:0-0:5'  'https://www.youtube.com/watch?v=OyuL5biOQ94'
...
$ ffprobe -v quiet -show_entries format=duration  Paradise\ -\ Coldplay\ \(Fingerstyle\ Guitar\)\ \[OyuL5biOQ94\].webm
[FORMAT]
duration=5.004000
[/FORMAT]
```

Reply:
Sorry for being picky with tests, reason is i really want to be able to trust the tests to catch breakage when auto updating yt-dlp

Reply:
will do,
but this gonna work with the sections that specified as a time range, not then chapter name specified:
```bash
yt-dlp --download-sections 'Intro'
```
Is there a way to get information on the length of the chapter?

Reply:
Maybe its in the Info.json but not sure. But just testing time range is fine with me

Reply:
Hey, got stuck on checking duration?